### PR TITLE
adds api driven documentation

### PR DESCRIPTION
**Pull request by stockholmux**
**Issue number:**

Closes # n/a

**Description of changes:**
Add conceptual documentation on bottlerocket's API.

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: content/en/os/1.13.x/concepts/api-driven/index.markdown [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+title = "API Driven"
+type = "docs"
+description = "Administration and setting manipulation with the Bottlerocket API" 
++++
+
+Bottlerocket is an API-driven operating system.
+This is a departure from general purpose Linux distributions where you install packages, configure services through individual configuration files, and use commands from the shell to perform administrative tasks.
+
+With the concept of variants, all installed software for any given image is a known quantity and an included API enables you to configure everything from a single interface.
+Additionally, administrative task like checking for updates and rebooting as well as executing commands on the host are all accomplished with API calls.
```
Comment (mchaker) [inline content/en/os/1.13.x/concepts/api-driven/index.markdown:11]: nit: task -> tasks?

File: content/en/os/1.13.x/concepts/api-driven/index.markdown [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+## Accessing the API
+
+The Bottlerocket API is only accessible by containers running on the same host with specific SELinux labels.
+The API is a Unix domain socket accessible by a mountable volume; containers running on the host or orchestrated containerd instance can access the API as long as they have both the correct SELinux labels and mount the socket.
```
Comment (mchaker) [inline content/en/os/1.13.x/concepts/api-driven/index.markdown:16]: nit: instance -> instances?

File: content/en/os/1.13.x/concepts/api-driven/index.markdown [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## API Client
+
+Bottlerocket ships with a tool called `apiclient` which provides a command line interface for interacting with the API.
+The client is a minimal abstraction over the API: it accepts command-line arguments, translates the arguments into an API call, and returns the response as standard output.
+Access to the `apiclient` binary is typically provided as a read-only mount from the host to containers that need access.
+
+## Settings
+
+You can view and alter Bottlerocket through the API.
```
Comment (mchaker) [inline content/en/os/1.13.x/concepts/api-driven/index.markdown:28]: nit: Bottlerocket -> Bottlerocket settings?

File: content/en/os/1.13.x/concepts/api-driven/index.markdown [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+Laterally, if you `get` a path and a setting, it returns only the value of that specific setting:
+
+```goat
+          +-- Path
+          |
++---------+---------+
+|                   | 
+host-containers.admin.enabled --> Value of: 
+                      |     |        eanbled
```
Comment (mchaker) [inline content/en/os/1.13.x/concepts/api-driven/index.markdown:67]: nit: typo: enabled

**Review by mchaker — CHANGES_REQUESTED** [review]
looks good, just a couple nits

File: content/en/os/1.13.x/concepts/api-driven/index.markdown (hunk 1 above)
Comment (stockholmux) [reply content/en/os/1.13.x/concepts/api-driven/index.markdown:11]: plurals are hard.

File: content/en/os/1.13.x/concepts/api-driven/index.markdown (hunk 2 above)
Comment (stockholmux) [reply content/en/os/1.13.x/concepts/api-driven/index.markdown:16]: Yup. And caught another issue in that sentence to. 🤦‍♂️

File: content/en/os/1.13.x/concepts/api-driven/index.markdown (hunk 3 above)
Comment (stockholmux) [reply content/en/os/1.13.x/concepts/api-driven/index.markdown:28]: Good catch. I should have been more accurate

File: content/en/os/1.13.x/concepts/api-driven/index.markdown (hunk 4 above)
Comment (stockholmux) [reply content/en/os/1.13.x/concepts/api-driven/index.markdown:67]: I wish the scheme had a property of 'eanbled' but alas...

**Pull request merged**